histogram.go

```diff
@@ -0,0 +1,42 @@
+package imaging
+
+import (
+	"image"
+)
+
+// Histogram returns a normalized histogram of an image.
+//
+// Resulting histogram is represented as an array of 256 floats, where
+// histogram[i] is a probability of a pixel being of a particular luminance i.
+func Histogram(img image.Image) [256]float32 {
```

Comment:
>Also what do you think about using slice or pointer to array to reduce copying of data?

Plain array is more clear though. Probably it doesn't make much sense to use slice/pointer here.